### PR TITLE
Coverity fixes vol. II

### PR DESCRIPTION
**Pull request by podhrmic**
This should fix the remaining airborne related issues from Coverity.
Not sure about the `switch` statements without `break`  - @flixr ?

### CODE REVIEW

**Comment by flixr** [comment]
The missing breaks are intended, it falls through to the common parts...

File: sw/airborne/math/pprz_algebra_int.c [hunk 1]
```diff
@@ -425,38 +425,43 @@ void int32_quat_of_rmat(struct Int32Quat *q, struct Int32RMat *r)
...
+                 (INT32_QUAT_FRAC - INT32_TRIG_FRAC + INT32_QUAT_FRAC - 1))
+                / two_qx;
+      }
     } else if (RMAT_ELMT(*r, 1, 1) > RMAT_ELMT(*r, 2, 2)) {
       const int32_t two_qy_two = RMAT_ELMT(*r, 1, 1) - RMAT_ELMT(*r, 0, 0)
                                  - RMAT_ELMT(*r, 2, 2) + TRIG_BFP_OF_REAL(1.);
       uint32_t two_qy = int32_sqrt(two_qy_two << INT32_TRIG_FRAC);
       two_qy = two_qy << (INT32_QUAT_FRAC - INT32_TRIG_FRAC);
+      if (two_qy != 0) {
       q->qi = ((RMAT_ELMT(*r, 2, 0) - RMAT_ELMT(*r, 0, 2)) <<
```
Comment (flixr) [inline sw/airborne/math/pprz_algebra_int.c]: indentation?

File: sw/simulator/nps/nps_main.c [hunk 2]
```diff
@@ -356,8 +356,7 @@ static bool nps_main_parse_options(int argc, char **argv)
           case 2:
             nps_main.fg_time_offset = atoi(optarg); break;
           case 3:
-            if (optarg == NULL) {nps_main.js_dev = strdup("0");}
-            else {nps_main.js_dev = strdup(optarg);}
+            nps_main.js_dev = strdup(optarg);
```
Comment (flixr) [inline sw/simulator/nps/nps_main.c]: does this still work? there must have been a reason for doing it this way...

File: sw/simulator/nps/nps_main.c [hunk 3]
```diff
@@ -369,24 +368,24 @@ static bool nps_main_parse_options(int argc, char **argv)
...
+          case 9:
             nps_main.fg_port_in = atoi(optarg); break;
         }
         break;
 
-      case 'j':
-        if (optarg == NULL) {nps_main.js_dev = strdup("0");}
-        else {nps_main.js_dev = strdup(optarg);}
-        break;
+          case 'j':
```
Comment (flixr) [inline sw/simulator/nps/nps_main.c]: unintended indentation changes?

File: sw/airborne/math/pprz_algebra_int.c (hunk 1 above)
Comment (podhrmic) [inline sw/airborne/math/pprz_algebra_int.c]: Fixed

File: sw/simulator/nps/nps_main.c (hunk 3 above)
Comment (podhrmic) [inline sw/simulator/nps/nps_main.c]: Silly Eclipse can't correctly indent nested switches. Fixed.

File: sw/simulator/nps/nps_main.c (hunk 2 above)
Comment (podhrmic) [inline sw/simulator/nps/nps_main.c]: Hmm, it doesn't or at least not the same way - marking as intentional, whatever the original reason here was...

Comment (flixr) [inline sw/simulator/nps/nps_main.c]: that's what I though... you can't make a copy of a NULL pointer, so in that case if it was not provided we explicitly use a string with `0` (as that is the intended default).

**Pull request merged**